x/distribution/client/cli/tx.go

```diff
@@ -102,20 +108,48 @@ $ %s tx distr withdraw-rewards exvaloper1alq9na49n9yycysh889rl90g9nhe58lcqkfpfg
 				return err
 			}
 
+			isVal := common.IsValidator(cliCtx, cdc, sdk.ValAddress(delAddr))
+			isDel := common.IsDelegator(cliCtx, cdc, delAddr)
+			needWarning := false
+
 			msgs := []sdk.Msg{}
-			if viper.GetBool(flagCommission) {
+			if viper.GetBool(flagCommission) || (isVal && !isDel) {
 				msgs = append(msgs, types.NewMsgWithdrawValidatorCommission(valAddr))
 			} else {
 				msgs = append(msgs, types.NewMsgWithdrawDelegatorReward(delAddr, valAddr))
+				if isVal && isDel {
+					needWarning = true
+				}
 			}
-
-			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, msgs)
+			result := transWrapError(common.NewWrapError(utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, msgs)), delAddr, valAddr)
+			if needWarning {
+				fmt.Fprintf(os.Stdout, "%s\n", fmt.Sprintf("\nFound address: \"%s\" is a validator, please add the `--commission` flag to the command line if you want to withdraw the commission, for example:\n"+
+					"%s ..... --commission.\n",
+					delAddr.String(), version.ClientName))
+			}
+			return result
 		},
 	}
 	cmd.Flags().Bool(flagCommission, false, "withdraw validator's commission")
 	return cmd
 }
 
+func transWrapError(wrapErr *common.WrapError, delAddr sdk.AccAddress, valAddr sdk.ValAddress) error {
```

Comment:
utAck.
cosmos-sdk/types/errors.Wrapf() function can add a description to a error. can we wrap a similar function or put the new wrapError to the  cosmos-sdk/types/errors?

Reply:
cosmos-sdk/types/errors is **desc**,but wrap is **log**
type Error struct {
	codespace string
	code      uint32
	desc      string
}

type WrapError struct {
	Code      uint32 `json:"code"`
	Log       string `json:"log"`
	Codespace string `json:"codespace"`
}